src/mpid/ch4/netmod/ucx/ucx_datatype.h

```diff
@@ -84,6 +84,9 @@ static inline ucs_status_t MPIDI_UCX_Unpack(void *state, size_t offset, const vo
 
     MPID_Segment_unpack(pack_state->segment_ptr, offset, &last, (void *) src);
 
+    if(unlikely (last == 0)){
```

Comment:
Right. This is a little tricky in the non-contig case. Looking at CH3, the strategy seems to be if we didn't successfully unpack everything, we set `MPI_ERR_TYPE` in the status to indicate a mismatch. https://github.com/pmodels/mpich/blob/master/src/mpid/ch3/channels/nemesis/netmod/mxm/mxm_poll.c#L324